packages/react/src/SubNav/SubNav.tsx

```diff
@@ -158,19 +158,24 @@ const _SubNavRoot = memo(({id, children, className, 'data-testid': testId, fullW
 type SubNavHeadingProps = {
   href: string
   'data-testid'?: string
+  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
 } & PropsWithChildren<React.HTMLProps<HTMLAnchorElement>> &
   BaseProps<HTMLAnchorElement>
```

Comment:
We might need to refactor this too? The element type is still a HTMLAnchorElement, but the root element isn't. This might create some confusion, particularly with the introduction of an `as` prop which specifies heading tags.

Reply:
What are you thinking it should be changed to?

The props get spread onto an `<a>` element regardless of what the `as` prop is set to, so I think `HTMLProps<HTMLAnchorElement>` is correct in this case.

Reply:
> The props get spread onto an <a> element regardless of what the as prop is set to

That's implementation detail. The user may find it confusing because the API now let's you do this:

```jsx
<SubNav.Heading href="#" as="h1" />
```

<h2> is now the new root element, not a link. Which in hindsight, how it probably should have been done in the first place 😅 

Happy to jam on some ideas with you over a call if easier..

in the meantime, some ideas:

- change to HTMLHeadingElement and spread the props into the root / heading instead
  - Makes more sense given the child is named `Heading`
  - add `href` as an optional, additional prop here, and leave a description / comment that it sets the href on the nested, optional link
  - This is gonna be a breaking change
  
- switch to `headingAs` and leave the types as is (not ideal, and I think sets a precedent so I think first option is better)

Reply:
Take a look at the approach in https://github.com/primer/brand/pull/709/commits/d46affe04f561eb0c3512c0ccb64a6ff2407348c

This leaves the rendering of the link up to the consumer, meaning we don't have to worry about supporting anchor link props now, or in the future. Consumers can add a `href`, `target`, `onClick`, `data-*` attribute, or anything else they like without us needing to modify the `SubNav.Heading` component.

```tsx
<SubNav.Heading as="h3">
  <a
    href="#"
    target="_blank"
    rel="noopener noreferrer"
    onClick={sendAnalytics}
    data-analytics="nav-heading"
    {/* or anything else they'd like to pass... */}
  >
    GitHub
  </a>
</SubNav.Heading>
```